lib/sablon/image.rb

```diff
@@ -0,0 +1,17 @@
+module Sablon
+  class Image
+    include Singleton
+    attr_reader :definitions
+
+    Definition = Struct.new(:name, :data, :rid) do
+      def inspect
+        "#<Image #{name}:#{rid}"
+      end
+    end
+
+    def self.create_by_path(path, random = nil)
```

Comment:
Does the `random = nil`and `"#{random || Random.new_seed}` only exist for testing purposes? (if not ignore the rest of this comment) If that is the case there may be a way to generated a fixed set of random integers so you can get a reproducible value without adding extra code. (I ran into this when testing a Python app awhile back)

A quick google search brought me here: https://silkandspinach.net/2008/02/01/tdd-and-random-numbers-in-ruby/